packages/custom-field/src/vaadin-custom-field-mixin.js

```diff
@@ -249,7 +249,20 @@ export const CustomFieldMixin = (superClass) =>
     /** @private */
     __setInputsFromSlot() {
       this._setInputs(this.__getInputsFromSlot());
-      this.__setValue();
+
+      // During initial render inputs might be empty due to the fact that
+      // slotted custom elements upgrade later than custom field, so they
+      // don't yet have `validate` method and don't pass "is input" check.
+      if (this.inputs.length === 0) {
+        return;
+      }
+
+      if (this.__storedValue) {
```

Comment:
There seems to be an option to avoid introducing `__storedValue` by using a different condition:

- If the number of inputs changes from 0 to > 0, propagate the component's value to the children: `this.__applyInputsValue(this.value)`
- Otherwise, propagate the children's value to the component: `this.__setValue()`

Reply:
Refactored the logic to use `__inputsOrValueChanged()` observer instead.